pkgs/applications/virtualization/xen/0004-fix-oxenstored-systemd-notification-race.patch

```diff
@@ -0,0 +1,15 @@
+diff -ru xen/tools/hotplug/Linux/launch-xenstore.in xen-delay1/tools/hotplug/Linux/launch-xenstore.in
+--- xen/tools/hotplug/Linux/launch-xenstore.in	2017-05-27 10:19:32.000000000 +0200
++++ xen-delay1/tools/hotplug/Linux/launch-xenstore.in	2017-05-29 23:33:53.000000000 +0200
+@@ -66,6 +66,11 @@
+ 
+ 	systemd-notify --booted 2>/dev/null || timeout_xenstore $XENSTORED || exit 1
+ 
++	# We wait for 10 seconds here as otherwise the sd_notify() notification
++	# from the oxenstored process might not be properly registered by systemd
++	# if this process quits before the notification is sent.
++	sleep 10
```

Comment:
It is hard to pick the right value here, as the call to `sd_notify()` may be delayed by an arbitrary amount of time. Probably 1 second would be enough, but I chose 10 just to be on a safe side. Please note that this delay will not delay the boot process, as systemd will go ahead and start other units as soon as it gets the notification. So the only effect of a longer delay will be the shell script running for some time afterwards. On the other hand, if the notification is lost then the startup of the unit will hang for a longer time until it gets timed out by systemd and fails.

This is an ugly workaround–I believe that you can solve it properly by using `MAINPID=` together with `sd_notify()`. I will submit a bug report about it to xen.